NuKeeper.BitBucket/BitbucketSettingsReader.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Linq;
+using NuKeeper.Abstractions;
+using NuKeeper.Abstractions.CollaborationPlatform;
+using NuKeeper.Abstractions.Configuration;
+
+namespace NuKeeper.BitBucket
+{
+    public class BitbucketSettingsReader : ISettingsReader
+    {
+        public Platform Platform => Platform.Bitbucket;
+
+        private static string Username { get; set; }
```

Comment:
So we need the username in our system to compose the basic auth header. We store it in a static string when the `RepositorySettings(Uri repositoryUri)`  is called and then get it out on the `UpdateCollaborationPlatformSettings(CollaborationPlatformSettings settings)`. Those two methods are from two different intances so that's why it's static.

My proposal would be to make all the ISettingsReader singleton when we injected them in the dependency container.